pkg/kubernetes/client/client.go

```diff
@@ -0,0 +1,214 @@
+limitations under the License.
+*/
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/container-storage-interface/spec/lib/go/csi"
+	"github.com/docker/go-units"
+	"github.com/openebs/jiva-csi/pkg/jivavolume"
+	"github.com/openebs/jiva-operator/pkg/apis"
+	jv "github.com/openebs/jiva-operator/pkg/apis/openebs/v1alpha1"
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+)
+
+const (
+	defaultReplicaCount = "3"
+	defaultReplicaSC    = "openebs-hostpath"
+	defaultNS           = "openebs"
+)
+
+// Client is the wrapper over the k8s client that will be used by
+// jiva-csi to interface with etcd
+type Client struct {
+	cfg    *rest.Config
+	client client.Client
+}
+
+// New creates a new client object using the given config
+func New(config *rest.Config) (*Client, error) {
+	c := &Client{
+		cfg: config,
+	}
+	err := c.Set()
+	if err != nil {
+		return c, err
+	}
+	return c, nil
+}
+
+// Set sets the client using the config
+func (cl *Client) Set() error {
+	c, err := client.New(cl.cfg, client.Options{})
+	if err != nil {
+		return err
+	}
+	cl.client = c
+	return nil
+}
+
+// RegisterAPI registers the API scheme in the client using the manager.
+// This function needs to be called only once a client object
+func (cl *Client) RegisterAPI() error {
+	mgr, err := manager.New(cl.cfg, manager.Options{})
+	if err != nil {
+		return err
+	}
+
+	// Setup Scheme for all resources
+	if err := apis.AddToScheme(mgr.GetScheme()); err != nil {
+		return err
+	}
+	return nil
+}
+
+// GetJivaVolume get the instance of JivaVolume CR.
+func (cl *Client) GetJivaVolume(name string) (*jv.JivaVolume, error) {
+	ns := "openebs"
+	instance := &jv.JivaVolume{}
+	err := cl.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: ns}, instance)
+	if err != nil {
+		logrus.Errorf("Failed to get JivaVolume CR: %v, err: %v", name, err)
+		return instance, err
+	}
+	return instance, nil
+}
+
+// UpdateJivaVolume update the JivaVolume CR
+func (cl *Client) UpdateJivaVolume(cr *jv.JivaVolume) error {
+	err := cl.client.Update(context.TODO(), cr)
+	if err != nil {
+		logrus.Errorf("Failed to update JivaVolume CR: %v, err: %v", cr.Name, err)
+		return err
+	}
+	return nil
+}
+
+func getDefaultLabels(pv string) map[string]string {
+	return map[string]string{
+		"openebs.io/persistent-volume": pv,
+		"openebs.io/component":         "jiva-volume",
+	}
+}
+
+// CreateJivaVolume check whether JivaVolume CR already exists and creates one
+// if it doesn't exist.
+func (cl *Client) CreateJivaVolume(req *csi.CreateVolumeRequest) error {
+	name := req.GetName()
+	sc := req.GetParameters()["replicaSC"]
+	rf := req.GetParameters()["replicaCount"]
+	ns, ok := req.GetParameters()["namespace"]
```

Comment:
Let's remove it from here, we will get the namespace from configclass

Reply:
What is configclass, is there any document that i can go through?